### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by hanpeter**
Thin wrapper for S3, just like [python-krux-boto-sqs](https://github.com/krux/python-krux-boto-sqs).

98% of the code in `krux_s3.s3.py` is copy and paste from [python-krux-manage-instance](https://github.com/krux/python-krux-manage-instance/blob/master/krux_s3/s3.py), and thus reviewed before.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,35 @@
 # krux_s3
+
+`krux_s3` is a library that provides wrapper functions for common S3 usage. It uses `krux_boto` to connect to AWS S3.
+
+## Warning
+
+In the current version, `krux_s3.s3.S3` is only compatible with `krux_boto.boto.Boto` object. Passing other objects, such as `krux_boto.boto.Boto3`, will cause an exception.
```
Comment (dcrampton) [inline README.md]: I think we need a couple more .s3 and .boto added on to the end of these.  Maybe six or seven more? ;)

File: README.md [hunk 2]
```diff
@@ -1 +1,35 @@
...
+    for key in s3.get_keys(bucket_name='my-test-bucket'):
+        print key.get_contents_as_string()
+
+### Run the application stand alone
+if __name__ == '__main__':
+    main()
+
+```
+
+As long as you get an instance of `krux_boto.boto.Boto`, the rest are the same. Refer to `krux_boto` module's [README](https://github.com/krux/python-krux-boto/blob/master/README.md) on various ways to instanciate the class.
```
Comment (dcrampton) [inline README.md]: Are the comments here specific to what you're creating, or are they from what you're building on top of?

File: krux_s3/s3.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+#
+# © 2015 Krux Digital, Inc.
+#
+
+# TODO: This is currently inside krux_manage_instance library.
+# However, consider breaking this into a separate library or add it to krux_boto library.
```
Comment (dcrampton) [inline krux_s3/s3.py]: I'm thinking you're doing this TODO right now.  Update in comments might be appropriate.

File: krux_s3/s3.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    )
+
+
+def add_s3_cli_arguments(parser, include_boto_arguments=True):
+    """
+    Utility function for adding S3 specific CLI arguments.
+    """
+    if include_boto_arguments:
+        # GOTCHA: Since EC2 and S3 both uses Boto, the Boto's CLI arguments can be included twice,
+        # causing an error. This creates a way to circumvent that.
```
Comment (dcrampton) [inline krux_s3/s3.py]: Good catch.

File: setup.py [hunk 5]
```diff
@@ -40,6 +40,7 @@
     ],
     entry_points={
         'console_scripts': [
+            'krux-s3-test = krux_s3.cli:main',
```
Comment (dcrampton) [inline setup.py]: What does the colon notation do here?

**Comment by dcrampton** [comment]
Assuming tested (what with the test bit at the bottom), looks good.

File: setup.py (hunk 5 above)
Comment (hanpeter) [inline setup.py]: It goes something like this: `'CLI_program_name = module(directory).module(sub-directory).module(file):function'`

File: README.md (hunk 1 above)
Comment (hanpeter) [inline README.md]: Like `krux_s3.s3._s3.s3_krux.really_s3.Final_S3`? :P

**Pull request merged**